### PR TITLE
fix: [Builder In World] Mark the entity as red in the "Entity List" when it is out of boundaries

### PR DESCRIPTION
**Pull request by sandrade-dcl**
This PR fix the issue: https://github.com/decentraland/unity-renderer/issues/148

image
image

### New `OnEntityBoundsCheckerStatusChanged` event in `SceneBoundsChecker`
This change includes the creation of a new event called `OnEntityBoundsCheckerStatusChanged` in the class `SceneBoundsChecker` that will allow us to check when an entity has entered or left to the boundaries.

**Testing link**: https://explorer.decentraland.zone/branch/fix/BiW-Mark-the-entity-as-red-in-the-Entity-List-when-it-is-outside-boundaries/index.html?ENV=org&ENABLE_BUILDER_IN_WORLD&position=100,100

### CODE REVIEW

**Review by D4rWiNSS — APPROVED** [review]
LGTM!

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/DecentralandEntity.cs [hunk 1]
```diff
@@ -32,8 +32,10 @@ public class DecentralandEntity : DCL.ICleanable, DCL.ICleanableEventDispatcher
         public System.Action<DCLTransform.Model> OnTransformChange;
         public System.Action<DecentralandEntity> OnMeshesInfoUpdated;
         public System.Action<DecentralandEntity> OnMeshesInfoCleaned;
+        public event Action<bool> OnEntityBoundsCheckerStatusChanged;
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/DecentralandEntity.cs]: Can't we have this event in `SceneBoundsChecker` itself? Having events for externals systems like this is kind of an anti-encapsulation pattern that grew over time.

I figure we can add some `Action<DecentralandEntity, bool>` that should do the trick? We want to listen for all the entities state change anyways.

**Review by BrianAmadori — COMMENTED** [review]
Is absolutely necessary to track this state adding new code to the already bloated `DecentralandEntity`?. I'd prefer to have this information decoupled and encapsulated in `SceneBoundsChecker` somehow.

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/DecentralandEntity.cs (hunk 1 above)
Comment (sandrade-dcl) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/DecentralandEntity.cs]: You're right, after our talk from yesterday I think it has more sense to move this responsability to the `SceneBoundsChecker` and not to overload more the `DecentralandEntity` class. You can see the change in this commit: https://github.com/decentraland/explorer/pull/2217/commits/30a9e6826f87eebdf89236b6696d7e48025d03cb Please take a quick look, thanks for the feedback, @BrianAmadori!

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/GodMode/Inspector/EntityListAdapter.cs [hunk 2]
```diff
@@ -96,4 +97,12 @@ void DeleteAdapter(DCLBuilderInWorldEntity entityToEdit)
             if (entityToEdit.entityUniqueId == currentEntity.entityUniqueId)
                 Destroy(gameObject);
     }
+
+    private void ChangeEntityBoundsCheckerStatus(DCL.Models.DecentralandEntity entity, bool isInsideBoundaries)
+    {
+        if (currentEntity.rootEntity.entityId != entity.entityId)
+            return;
+
+        nameInputField_Text.color = isInsideBoundaries ? entityInsideOdBoundsColor : entityOutOfBoundsColor;
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/GodMode/Inspector/EntityListAdapter.cs:106]:
```suggestion
        nameInputField_Text.color = isInsideBoundaries ? entityInsideOfBoundsColor : entityOutOfBoundsColor;
```
Typo in `entityInsideOdBoundsColor`.

**Review by BrianAmadori — APPROVED** [review]
LGTM! ty for applying the feedback!

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/GodMode/Inspector/EntityListAdapter.cs (hunk 2 above)
Comment (sandrade-dcl) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/GodMode/Inspector/EntityListAdapter.cs:106]: Ups, good catch! I will fix it in my other PR (https://github.com/decentraland/explorer/pull/2224) where I'm already modifying the `EntityListAdapter` prefab. Thanks!

**Pull request merged**